packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,174 @@
+                .then(fontData => {
+                    if (fontData) {
+                        this._injectFontRawStyle(fontData);
+                        this._overrideCorePlayerFontStyles(data.fontFamily);
+                        this._loadedFontName = data.downloadData.name;
+                    }
+                    //todo [sa] should we override core player font style even if we don't have the font ???
+                })
+                .catch(err => {
+                    logger.error(`Failed to download font`, {
+                        method: "loadFont -> _downloadFont",
+                        data: {
+                            error: err
+                        }
+                    });
+                    //todo [sa] should we override core player font style even if we don't have the font ???
+                });
+        } else {
+            // no download data exists, only override core player font style
+            this._overrideCorePlayerFontStyles(data.fontFamily);
+        }
+    }
+
+    private _overrideCorePlayerFontStyles(fontFamily: string): void {
+        const fontCss = `.kaltura-player-container {
+                font-family: inherit;
+            }         
+            .playkit-player {
+                font-family: ${fontFamily};
+            }
+            .playkit-player .playkit-player-gui  {
+                font-family: ${fontFamily};
+            }
+            button, textarea {
+                font-family: inherit;
+            }`;
+        const lastHeadChild = (document.head || document.getElementsByTagName("head")[0])
+            .lastElementChild;
+        const style = document.createElement("style");
+        //adding as last child
+        lastHeadChild.parentNode.insertBefore(style, lastHeadChild.nextSibling);
+        style.appendChild(document.createTextNode(fontCss));
+    }
+
+    private _checkFontExistence(fontName): boolean {
+        // creating an in-memory Canvas element
+        const canvas = document.createElement("canvas");
+        const context = canvas.getContext("2d");
+        // the text whose final pixel size we want to measure
+        const text = "abcdefghiiiiiiiiijklmnopqrstuvwwwwwwwwwwxyz0123456789";
+        // baseline font
+        context.font = `72px monospace`;
+        // size of the baseline text
+        const baselineSize = context.measureText(text).width;
+        // specifying the font whose existence we want to check
+        context.font = `72px ${fontName},monospace`;
+        // checking the size of the font we want to check
+        const newSize = context.measureText(text).width;
+        // If the size of the two text instances is the same, the font does not exist
+        // because it is being rendered using the same font
+        return newSize !== baselineSize;
+    }
+
+    private _downloadAndCacheFont(url: string, fontName: string): Promise<string | null> {
+        return axios.get(url).then((result: any) => {
```

Comment:
can `axios.get` result with Promise.catch?

Reply:
not relevant